### PR TITLE
Feature/syl 170 bancontact new payplug gateway

### PR DESCRIPTION
**Pull request by tuyetrinhvo**

### CODE REVIEW

File: src/Checker/CanSaveBancontactMethodChecker.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+namespace PayPlug\SyliusPayPlugPlugin\Checker;
+
+use PayPlug\SyliusPayPlugPlugin\ApiClient\PayPlugApiClientInterface;
+use PayPlug\SyliusPayPlugPlugin\Gateway\BancontactGatewayFactory;
+
+final class CanSaveBancontactMethodChecker
+{
+    /** @var PayPlugApiClientInterface */
+    private $client;
```
Comment (oallain) [inline src/Checker/CanSaveBancontactMethodChecker.php:13]: Can you add typehint in properties and remove phpDoc ?

File: src/Gateway/Form/Type/BancontactGatewayConfigurationType.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                    new NotBlank([
+                        'message' => 'payplug_sylius_payplug_plugin.secret_key.not_blank',
+                    ]),
+                    new IsPayPlugSecretKeyValid(),
+                    new IsCanSaveBancontactMethod(),
+                ],
+                'help' => $this->translator->trans('payplug_sylius_payplug_plugin.ui.retrieve_secret_key_in_api_configuration_portal'),
+                'help_html' => true,
+            ])
+            ->add(PayPlugGatewayFactory::ONE_CLICK, CheckboxType::class, [
```
Comment (oallain) [inline src/Gateway/Form/Type/BancontactGatewayConfigurationType.php:44]: IMO, we don't have One Click on Bancontact, right ?

**Review by oallain — CHANGES_REQUESTED** [review]

File: src/Gateway/Form/Type/BancontactGatewayConfigurationType.php (hunk 2 above)
Comment (tuyetrinhvo) [reply src/Gateway/Form/Type/BancontactGatewayConfigurationType.php:44]: They are removed in next PR

File: src/Gateway/Validator/Constraints/IsCanSaveBancontactMethod.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @Annotation
+ */
+final class IsCanSaveBancontactMethod extends Constraint
+{
+    /** @var string */
+    public $noTestKeyMessage = 'payplug_sylius_payplug_plugin.bancontact.can_not_save_method_with_test_key';
```
Comment (oallain) [inline src/Gateway/Validator/Constraints/IsCanSaveBancontactMethod.php:15]:
```suggestion
    public string $noTestKeyMessage = 'payplug_sylius_payplug_plugin.bancontact.can_not_save_method_with_test_key';
```

File: src/Gateway/Validator/Constraints/IsCanSaveBancontactMethod.php [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    /** @var string */
+    public $noAccessMessage = 'payplug_sylius_payplug_plugin.bancontact.can_not_save_method_no_access';
```
Comment (oallain) [inline src/Gateway/Validator/Constraints/IsCanSaveBancontactMethod.php:18]:
```suggestion
    public string $noAccessMessage = 'payplug_sylius_payplug_plugin.bancontact.can_not_save_method_no_access';
```

**Review by oallain — CHANGES_REQUESTED** [review]

**Review by oallain — APPROVED** [review]

**Review by TheGrimmChester — APPROVED** [review]

**Pull request merged**